### ISSUE TITLE
Deployment failed - how to troubleshoot?

### ISSUE THREAD

**Issue by ffxsam** [arunoda/meteor-up-legacy#447]
Here's my scenario:

I'm testing out using mup to install several apps onto one Ubuntu 14.04 server. I started by deploying the "todos" sample project. I ran `mup init`, configured the mup.json file to install everything needed. Ran `mup setup` followed by `mup deploy`, and no problems at all. The app is running on the server, port 3000.

So now I tried to deploy the leaderboard sample app. Same mup.json file pretty much, except I set the value `false` on variables for setting up node/mongo/phantom since they're already set up. I skipped the `mup setup` step, and ran `mup deploy`. I get this:

```
Started TaskList: Deploy app 'leaderboard' (linux)
[192.168.11.50] - Uploading bundle
[192.168.11.50] x Uploading bundle: FAILED
    No such file
```

I wish there were some sort of details so I could see specifically what went wrong. There's nothing in the mup log on the server since it apparently didn't get that far. Any suggestions? Should I have run `mup setup` anyway?

My plan is to set up leaderboards on port 3001, and then set up nginx so that todos and leaderboards are vhosts, with nginx doing a `proxy_pass` to ports 3000 and 3001, respectively.

**Comment by arunoda** [comment]
You need to do mup setup again.

**Comment by ffxsam** [comment]
Aha! Thanks! And do I need to set `MONGO_URL` to something specific? Or will it automatically use the server that's already installed, and create a db for the app?

**Comment by arunoda** [comment]
If you enabled setupMongo true in mup json. It will create a DB for your
app and it will be used automatically.

You can override if with the custom mongodb DB by setting 'MONGO_URL'

**Comment by ffxsam** [comment]
When using mup a second time (on the same server), do I need to set `setupNode` and `setupMongo` to false? Or leave them set to true, and mup will check if they're already installed and skip them?

**Comment by romans127** [comment]
What if I already deployed an app, and I want to deploy another one.

**Comment by ffxsam** [comment]
You'd have to have an environment set up where you're running nginx and using `proxy_pass` to direct different URLs to different internal port numbers.

**Comment by romans127** [comment]
yes I have that setup I have all my meteor apps coming in under port 4000, and I have a coldfusion app on 3000.

**Comment by romans127** [comment]
I'm just concerned if I deploy app1 , then deploy app2 if it will destroy the app1 ?

**Comment by ffxsam** [comment]
Nope, I successfully have three Meteor apps on a single server. Just make sure that in `mup.json`, app2 is named totally differently from app1, and you're setting the port differently. (I use port 3000 for app1, 3001 for app2, etc)

**Comment by romans127** [comment]
ah i see, ok so how do I dynamically tell nginx to route the port. I don't want to modify nginx conf file every time. I deploy an app since i'm following a convention by sending all of my apps within a specific port range. 

If you could instruct on how to have nginx serve any app with the following credentials. PS I have already tried `$request_port`

all of my meteor apps will be served with the `*.app.mydomain.com`

```
//App1
app1.app.mydomain.com:4001

//app2
app2.app.mydomain.com:4002

```

**Comment by ffxsam** [comment]
As far as I know, there's no dynamic way. I have an nginx conf file for every app, and within each conf file it points to a different internal port.

**Comment by romans127** [comment]
I see

**Comment by hrhitesh** [comment]
@ffxsam i have problem when i redeploy app2 on 3000 instead of app1, it shows me always app1. How should i overcome with this situation?
